src/Main.java

```diff
@@ -10,186 +10,156 @@
-            else break;
+        int chunks = (int) Math.ceil((double) bytes.length() / CHUNK_SIZE);
+        bytes += "0".repeat(CHUNK_SIZE * chunks - bytes.length());
+
+        int[] hValues = new int[NUM_PRIMES];
+        for (int i = 0; i < NUM_PRIMES; i++) {
+            hValues[i] = dConstants.get((chunks + i) % NUM_PRIMES);
         }
-        bytes += "0".repeat(32*chunks - (bytes.length()));
-        
-        Integer h0,h1,h2,h3,h4,h5,h6,h7;
-        h0 = DCONSTANTS.get(chunks % 10);
-        h1 = DCONSTANTS.get((chunks+1) % 10);
-        h2 = DCONSTANTS.get((chunks+2) % 10);
-        h3 = DCONSTANTS.get((chunks+3) % 10);
-        h4 = DCONSTANTS.get((chunks+4) % 10);
-        h5 = DCONSTANTS.get((chunks+5) % 10);
-        h6 = DCONSTANTS.get((chunks+6) % 10);
-        h7 = DCONSTANTS.get((chunks+7) % 10);
-        //System.out.println(h0);
+
         for (int i = 0; i < chunks; i++) {
-            Integer a,b,c,d,e,f,g,h;
-            a=h0;
-            b=h1;
-            c=h2;
-            d=h3;
-            e=h4;
-            f=h5;
-            g=h6;
-            h=h7;
-            ArrayList<Integer> w = new ArrayList<>();
-            for (int j = (bytes.length()/chunks*i)/4; j < ((bytes.length()/chunks)*(i+1))/4; j++) {
-                char[] chars = new char[4];
-                bytes.getChars(4*j,4+4*j,chars,0);
-                String newStr = new String(chars);
-                //System.out.println(newStr);
-                w.add(stringToBytesToInt(newStr));
-            }
+            int[] abcdefgh = new int[NUM_PRIMES];
+            System.arraycopy(hValues, 0, abcdefgh, 0, NUM_PRIMES);
+            List<Integer> w = getChunkWords(bytes, i, chunks);
+
             for (int j = 8; j < 64; j++) {
-                int s0, s1, s2, ssum;
-                s0 = w.get(j-1)^(w.get(j-3)%4);
-                s1 = w.get(j-2) ^ (w.get(j-6) % (3));
-                s2 = s0 + (s1);
-                ssum = s2 + (w.get(j-7));
+                int s0 = w.get(j - 1) ^ (w.get(j - 3) % 4);
+                int s1 = w.get(j - 2) ^ (w.get(j - 6) % 3);
+                int s2 = s0 + s1;
+                int ssum = s2 + w.get(j - 7);
                 w.add(ssum);
             }
+
             for (int j = 0; j < 64; j++) {
-                int abc, cd, efg, temp1, temp2;
-                int index = (3*j+1) % rf.size();
+                int index = (3 * j + 1) % rf.size();
                 BinaryOperator<Integer> fn = rf.get(index);
-                //System.out.println(j);
-                abc = a ^ (fn.apply(b, c));
-                cd = d ^ (fn.apply(e, a));
-                efg = f ^ (fn.apply(a, g));
-                temp1 = abc + (cd + (w.get(j)));
-                temp2 = efg + (w.get(j));
-                h=g;
-                g=d + (temp1);
-                f=a + (temp1 + (temp2));
-                e=d;
-                d=c;
-                c=b + (temp2);
-                b=a;
-                a=h;
+                int a = abcdefgh[0];
```

Comment:
Lines 94-103: I define abc, cd, efg for a reason. Without it you also fuck up my temp1 and temp2.